resolwe/flow/views.py

```diff
@@ -182,6 +204,12 @@ def perform_create(self, serializer):
             collection = Collection.objects.get(pk=c)
             collection.data.add(instance)
 
+        ds_slug = self.request.data.get('descriptor_schema', None)
+        if ds_slug:
+            descriptor_schema = DescriptorSchema.objects.filter(slug=ds_slug).order_by('version').last()
+            instance.descriptor_schema = descriptor_schema
+            instance.save()
```

Comment:
`DescriptorSchema` is added in `perform_create`. It would be probably better if `descriptor_schema` is passed to `DataSerializer` and it handles it on initial save.